src/Illuminate/Foundation/Bootstrap/HandleExceptions.php

```diff
@@ -194,17 +194,23 @@ public function handleException(Throwable $e)
     {
         self::$reservedMemory = null;
 
+        $uncaught = false;
+
         try {
             $this->getExceptionHandler()->report($e);
         } catch (Exception $e) {
-            //
+            $uncaught = true;
         }
 
         if (static::$app->runningInConsole()) {
             $this->renderForConsole($e);
         } else {
             $this->renderHttpResponse($e);
         }
+
+        if ($uncaught) {
+            exit(1);
```

Comment:
Wouldn't this cause Octane to exit the worker on every uncaught exception, which is not desired?

Reply:
good question. That could be the case yes.

Anyhow, there is no way to say to php "exit with exitCode=1 on your own accord (when its time to exit)". AFAIK you *must* call `exit(1)` yourself which will then still call any remaining `__destruct()` listeners and shutdown handlers.

Reply:
I moved the `exit(1)` call a few lines up now (inside the `$app->runningInConsole()` since this is a command line related issue)

Reply:
Octane workers still will have `$app->runningInConsole() === true` since it is an artisan command.